src/niri.rs

```diff
@@ -806,7 +808,15 @@ impl State {
                 }
             }
 
-            if self.niri.config.borrow().input.keyboard.track_layout == TrackLayout::Window {
+            if self
+                .niri
+                .config
+                .borrow()
+                .input
+                .default_keyboard()
```

Comment:
I guess we should track this per-keyboard? Then should also be careful which layout to switch to in `on_keyboard()` when switching keyboards.

Reply:
hrm not sure i have thought this through completely :thinking:

Reply:
> Then should also be careful which layout to switch to in `on_keyboard()` when switching keyboards.

meaning we should only switch to `Global` layouts in `on_keyboard()`?

Reply:
If you type with a track_layout = window keyboard, then type with a second keyboard, then switch windows, then type with the first keyboard, it should pick the correct layout for the first keyboard taking the window switch into account.

Reply:
new changes address the `on_keyboard()` handling, and it seems to work in my tests. i might be missing something, but i'm not currently tracking any state per-keyboard.